CONTRIBUTING.md

```diff
@@ -109,6 +106,66 @@ The contribution process is outlined below:
    maintainer. At this point, a Meta employee will be notified to merge your PR,
    due to tooling limitations.
 
+## Commit Messages
+
+We build Velox for the long-run, and to do so it is crucial that project
+maintainers are able to efficiently inspect and understand project logs.
+
+Commit messages that follow a strict pattern improve maintainability, and allow
+tasks such as summarizing changelogs and identifying API breaking changes to be
+automated. Despite requiring more rigor from authors and reviewers, consistency
+creates productivity in the long term.
+
+In Velox, commit messages are generated based on the input provided to PRs, and
+must follow the [conventional commit
+specification](https://www.conventionalcommits.org/en/v1.0.0/) in the following
+manner:
+
+**PR titles** must follow the pattern:
+
+> \<type\>[optional scope]: \<description\>
+
+where:
+
+* *Type* can be any of the following keywords:
+  * **feat** when new features are being added.
+  * **fix** for bug fixes.
+  * **build** for build or CI-related improvements.
+  * **docs** for enhancements to documentation (only).
+  * **refactor** for refactoring (no logic changes).
+
+* PR titles also take an *optional scope* field containing the area
+  of the code being target by the PR, to further help commit classification.
+  For example, "fix(expression): " or "refactor(parquet): " or "feat(memory):".
+  It is ok to omit this field if there is no adequate classification for the
+  PR, or if the PR touches multiple different components.
+
+* A *description* sentence summarizing the PR, written in imperative tone. The
+  description must be capitalized, not contain a period at the end, and be kept
+  under a 80 characters limit.
```

Comment:
> under a 80 characters limit.

I believe the whole commit title needs to be within that limit, not just the "Description" sentence.

"refactor(expression)" takes up 20 characters, 1/4 of the limit. Is this a concern?

Reply:
> Is this a concern?

I believe that's why conventional commit choose short words (`chore` vs `refactor`) or abbreviations like `feat`

Reply:
Users can leverage the conventional commit words to keep the title succinct.
`Add support for IPPREFIX type`
vs.
`feature(type): Add IPPREFIX`

Reply:
+1. They do add extra characters, but like @majetideepak pointed out, these are usually things you would have in the title anyway. For example:

> Refactor expression code to prepare for feature X

could be:

> refactor(expr): Prepare for feature X

I think in general abbreviating things that are very intuitive like `expression` -> `expr` should also be ok. 

Btw, @mbasmanova, when you look at these title example individually, the value is not super clear, but if you have a list of hundreds of them already properly formatted (vs the free form version) it makes a ton of difference.